### PR TITLE
feat: CI add deno formatting

### PR DESCRIPTION
**Pull request by edisontim**
# Adds formatting in CI with deno for indexer
Time spent on this PR: 30 mins

Resolves: #1147

# Pull Request type
Please check the type of change your PR introduces:

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

deno formatting runs on pushes and pull requests

# Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: .github/workflows/indexer_formatting.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
```
Comment (greged93) [inline .github/workflows/indexer_formatting.yml:1]: Could we rename this "fmt" and just move the rust `lint` job here too?

File: .github/workflows/indexer_formatting.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  deno-formatting:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          sparse-checkout: indexer
+      - name: Install zip
+        run: sudo apt update && sudo apt upgrade && sudo apt install zip
```
Comment (greged93) [inline .github/workflows/indexer_formatting.yml:17]: why is this needed?

Comment (edisontim) [reply .github/workflows/indexer_formatting.yml:17]: Deno installation requires the `unzip` command which comes with the `zip` package. `zip` isn't installed by default on the image of the CI

**Review by greged93 — APPROVED** [review]
